backpack-compose/tokens.gradle.kts

```diff
@@ -0,0 +1,65 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018-2021 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import net.skyscanner.backpack.tokens.*
+
+tasks {
+
+  val tokensPackage = "net.skyscanner.backpack.compose.tokens"
+  val group = "tokens"
+  val src = project.projectDir.resolve("src/main/kotlin").path
+
+  val source = project.nodeFileOf("@skyscanner/bpk-foundations-android", "tokens/base.raw.android.json")
+    .readAs(BpkFormat.Json)
+
+  val elevationTokens by creating {
+    this.group = group
+    doLast {
+      source
+        .parseAs(BpkDimension.Category.Elevation)
+        .transformTo(BpkDimension.Format.Compose(namespace = "BpkElevation"))
+        .saveTo(BpkOutput.KotlinFile(src, tokensPackage))
+        .execute()
+    }
+  }
+
+  val spacingTokens by creating {
+    this.group = group
+    doLast {
+      source
+        .parseAs(BpkDimension.Category.Spacing)
+        .transformTo(BpkDimension.Format.Compose(namespace = "BpkSpacing"))
+        .saveTo(BpkOutput.KotlinFile(src, tokensPackage))
+        .execute()
+    }
+  }
+
+  val radiiTokens by creating {
+    this.group = group
+    doLast {
+      source
+        .parseAs(BpkDimension.Category.Radii)
+        .transformTo(BpkDimension.Format.Compose(namespace = "BpkBorderRadius"))
+        .saveTo(BpkOutput.KotlinFile(src, tokensPackage))
+        .execute()
+    }
+  }
+
+  val sizeTokens by creating {
```

Comment:
should we generate a `BpkDimension` file as well to contain all the different types of dimension tokens for easier discoverabilit?

Reply:
Just having it in one package should be discoverable enough. If we do nested object, notations like BpkDimenson.BpkElevation.Sm may be harder to read and write. I think people will get used to the dimensions quite quickly, considering we also have autosuggest in IDE.

Reply:
I'm not sure the package + auto complete is discoverable enough - there'll be many things starting with `Bpk` but you won't easlily know what's a dimension, rather than being able to use `BpkDimension.x` - you can also add imports for `BpkElevation` directly for the common ones or the ones you know

Reply:
It's pretty much similar to tokens naming in XMLs right now (e.g. bpkElevationXs). Should be very intuitive to find, providing that we'll also have examples on the website. If not, we can always change it later :)

Reply:
When I type @dimen/something, my focus is on "something". I already know type I'm looking for.

What we can do, however, is manually create a bridge (e.g. keeping original classes intact). Something like:

```
object BpkDimension {

   val Elevation = BpkElevation

   val Spacing = BpkSpacing

}
```

But to me honest, I've never seen such notation is used in Compose. People just put constants in objects, like here:
https://github.com/androidx/androidx/blob/androidx-main/compose/material/material/src/commonMain/kotlin/androidx/compose/material/Button.kt#L290

Reply:
Yep that's what I originally proposed :) 

these dimensions are more our global dimensions than tied to a specific component like a button where it makes sense to keep them scoped to the button (and keep in the same file as Button as well)